### PR TITLE
Betting 404 -- bet classes and match algorithm

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/betting/betting_math.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+{
+    FC_ASSERT(bet1_stake.symbol() == SCORUM_SYMBOL && bet1_stake.symbol() == bet2_stake.symbol(),
+              "Invalid sumbol for stake");
+    FC_ASSERT((odds_fraction_type)bet1_odds == bet2_odds.inverted()
+                  && (odds_fraction_type)bet2_odds == bet1_odds.inverted(),
+              "Odds for bet 1 ans bet 2 must make 1 in summ of it's probability");
+
+    matched_stake_type result;
+
+    result.bet1_potential_result = bet1_stake * bet1_odds;
```
Comment (Katkevich) [inline libraries/chain/betting/betting_math.cpp]: It is enough to return matched stakes for bet1/bet2. So this code can be simplified:
```
matched_stake_type result;
result.bet1_stake = bet1_stake;
result.bet2_stake = bet2_stake;

auto bet1_result = bet1_stake * bet1_odds;
auto bet2_result = bet2_stake * bet2_odds;

if (bet1_result > bet2_result)
{
    result.bet1_stake = bet2_result * odds_fraction_type(bet1_odds).coup();
    result.bet1_stake = std::max(result.bet1_stake, asset(1, bet1_stake.symbol()));
}
else if (bet1_result < bet2_result)
{
    result.bet2_stake = bet1_result * odds_fraction_type(bet2_odds).coup();
    result.bet2_stake = std::max(result.bet2_stake, asset(1, bet2_stake.symbol()));
}
```

File: tests/utests/betting/betting_matcher_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+BOOST_FIXTURE_TEST_SUITE(betting_matcher_tests, betting_matcher_fixture)
+
+SCORUM_TEST_CASE(matching_not_found_and_created_pending_check)
+{
+    const auto& new_bet = create_bet();
+
+    service.match(new_bet);
+
+    BOOST_CHECK(pending_bet.is_exists());
```
Comment (alexplevako) [inline tests/utests/betting/betting_matcher_tests.cpp]: I suppose this is `pending_bets` and it checks that the list not empty?
```cpp
BOOST_CHECK(!pending_bets.empty());
```

File: tests/utests/betting/betting_matcher_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    BOOST_CHECK(!matched_bet.is_exists());
```
Comment (alexplevako) [inline tests/utests/betting/betting_matcher_tests.cpp]:
```cpp
BOOST_CHECK(matched_bets.empty());
```

**Comment by alexplevako** [comment]
use squash and merge.

**Pull request merged**